signinapp/event.py

```diff
@@ -59,26 +70,39 @@ def selfevent():
         flash("Event is not active")
         return redirect(url_for("index"))
 
-    if not current_user.is_signed_into(ev):
-        ev.scan(current_user.code)
+    if current_user.is_signed_into(ev):
+        active: Active = db.session.scalar(
+            select(Active).filter_by(user=current_user, event=ev)
+        )
+        active.convert_to_stamp()
 
-    return render_template("selfscan.html.jinja2", event=ev, event_code=ev.code)
+    return redirect(url_for("index"))
 
 
 @eventbp.route("/scan", methods=["POST"])
 def scan():
     event = request.values.get("event_code")
     user_code = request.values.get("user_code")
 
-    ev: Event = db.session.scalar(select(Event).filter_by(code=event))
+    if not user_code:
+        return Response(
+            f"Error: Not a valid QR code: {user_code}", HTTPStatus.BAD_REQUEST
+        )
+    if not (user := User.from_code(user_code)):
+        return Response("Error: User does not exist", HTTPStatus.BAD_REQUEST)
+
+    if not user.approved:
+        return Response("Error: User is not approved", HTTPStatus.BAD_REQUEST)
```

Comment:
403 here?

Reply:
I did 401 - from the `HTTPStatus` definition, 403 is described as "authorization would not help"